zerver/lib/stream_topic.py

```diff
@@ -32,5 +36,22 @@ def user_ids_muting_topic(self) -> Set[int]:
             for row in query
         }
 
+    def get_user_ids_muting_stream(self) -> Set[int]:
+        subscription_rows = Subscription.objects.filter(
+            recipient__type=Recipient.STREAM,
+            recipient__type_id=self.stream_id,
+        ).distinct('user_profile_id').values('user_profile_id', 'is_muted')
+
+        return {
+            row['user_profile_id']
+            for row in subscription_rows
+            if row['is_muted']
+        }
+
+    def get_contributors_for_topic(self, recipient: Recipient) -> QuerySet:
+        return messages_for_topic(
+            recipient.id, self.topic_name
+        ).distinct('sender_id')
```

Comment:
Maybe worth adding a comment that we currently define contributors to be users who've sent a message to the topic, but we might later extend it to include users who've reacted to it (or marked themselves as following it).

Reply:
For that reason I'd argue we should call this `get_followers_user_ids_for_topic`; I think that's what we'll call this in the future.